staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -622,3 +528,68 @@ func TestInvalidKeys(t *testing.T) {
 	_, countErr := store.Count(invalidKey)
 	expectInvalidKey("Count", countErr)
 }
+
+func BenchmarkStore_GetList(b *testing.B) {
+	testCases := []struct {
+		name      string
+		objectNum int
+	}{
+		{
+			name:      "list 10 objects",
+			objectNum: 10,
+		},
+		{
+			name:      "list 100 objects",
+			objectNum: 100,
+		},
+		{
+			name:      "list 1000 objects",
+			objectNum: 1000,
+		},
+		{
+			name:      "list 10000 objects",
+			objectNum: 10000,
+		},
+		{
+			name:      "list 50000 objects",
+			objectNum: 50000,
+		},
+	}
+
+	for _, tc := range testCases {
+
+		b.Run(tc.name, func(b *testing.B) {
+
+			// booting etcd instance
+			ctx, store, etcdClient := testSetup(b)
+			defer etcdClient.Close()
+
+			// make fake pods..
+			dir := "/testing"
+			key := dir + "/testkey"
+			originalRevision := ""
+			for i := 0; i < tc.objectNum; i++ {
+				out := &example.Pod{}
+				obj := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("pod-%d", i), SelfLink: "testlink"}}
```

Comment:
Hmm - this is fairly small object.
Can we try having some more-representative object?

TBH, I'm a bit surprised that allocating a new slice (even though it's just pointers) is actually universally faster...

Reply:
> Can we try having some more-representative object?

I checked, examplev1.Pod is indeed relatively small. What do you think of corev1.Pod and corev1.Node?

Reply:
Fixed,  PTAL @wojtek-t